src/components/AccountSeed.js

```diff
@@ -17,154 +17,93 @@
-    this.getWordPosition = this.getWordPosition.bind(this);
-    this.keyboardDidShow = this.keyboardDidShow.bind(this);
-    this.keyboardDidHide = this.keyboardDidHide.bind(this);
-  }
-
-  keyboardDidShow() {
-    this.setState({ keyboard: true, animation: true });
-    Animated.timing(
-      // Animate over time
-      this.state.suggestionsHeight, // The animated value to drive
-      {
-        toValue: 35, // Animate to opacity: 1 (opaque)
-        duration: 200 // Make it take a while
-      }
-    ).start(() => this.setState({ animation: false }));
-  }
-
-  keyboardDidHide() {
-    this.setState({ animation: true });
-    Animated.timing(
-      // Animate over time
-      this.state.suggestionsHeight, // The animated value to drive
-      {
-        toValue: 0, // Animate to opacity: 1 (opaque)
-        duration: 200 // Make it take a while
-      }
-    ).start(() => {
-      this.setState({ keyboard: false, animation: false });
-    });
-  }
-
-  handleCursorPosition({
-    nativeEvent: {
-      selection: { start, end }
-    }
-  }) {
-    if (start !== end) {
-      return;
-    }
-    this.setState({ cursorPosition: start });
-  }
-
-  getSearchInput() {
-    const { value } = this.props;
-    const { cursorPosition } = this.state;
-    let startFrom = cursorPosition;
-    // find the first space or start of string
-    while (startFrom > 0 && [' '].indexOf(value.charAt(startFrom - 1)) === -1) {
-      --startFrom;
-    }
-    return value.substring(startFrom, cursorPosition);
-  }
-
-  getWordPosition() {
-    const { value } = this.props;
-    const { cursorPosition } = this.state;
-    let wordPosition = 0;
-    let cursor = 0;
-    let char = '';
-    let wasLetter = false;
-    if (0 === value.length) {
-      return 0;
-    }
-    while (true) {
-      if (cursorPosition === cursor || (char = value.charAt(cursor)) === '') {
-        break;
-      }
-      if ([' ', '\n', '\r'].indexOf(char) === -1) {
-        wasLetter = true;
-      } else {
-        if (wasLetter) {
-          ++wordPosition;
-        }
-        wasLetter = false;
-      }
-      ++cursor;
-    }
-    return wordPosition;
-  }
-
-  getSuggestions(input, words) {
-    let fromIndex = WORDS.findIndex(w => w.startsWith(input));
+  getSuggestions = (inputWordArray, word_list) => {
+    const input = inputWordArray[inputWordArray.length - 1]
+    // the word list is sorted, get the index we should start searching
+    // for the word
+    let fromIndex = word_list.findIndex(w => w.startsWith(input));
     if (fromIndex === -1) {
       return [];
     }
-    if (WORDS[fromIndex] === input) {
-      fromIndex = 0;
-    }
+
     const SUGGESTIONS_COUNT = 5;
     const result = [];
     let yielded = 0;
-    result.push(WORDS[fromIndex]);
-    while (yielded < SUGGESTIONS_COUNT && WORDS[fromIndex] !== undefined) {
-      ++fromIndex;
-      if (!WORDS[fromIndex].startsWith(input)) {
+    while (yielded < SUGGESTIONS_COUNT && fromIndex < word_list.length) {
```

Comment:
I'd suggest doing:

```js
const previousWords = inputWordArray.slice(0, -1); // input without last word
const suggestions = wordList.slice(fromIndex + 1, fromIndex + SUGGESTIONS_COUNT));

const result = [wordList[fromIndex]];

for (const suggestion of suggestions) {
  if (!suggestion.startsWith(input)) {
    break;
  }

  // do not suggest words that where already used
  if (previousWords.indexOf(suggestion) === -1) {
    result.push(suggestion);
  }
}

return result;
```

No magic integers tracking things.

Reply:
So much better thanks.
Just a nit, didn't test, I guess we should actually have something like:
`wordList.slice(fromIndex, fromIndex + SUGGESTIONS_COUNT));`
to start at `fromIndex` and have the right amount of `SUGGESTIONS_COUNT`.

Also because of the last `if` statement, we might end up not suggesting the amount of words we originally wanted but that probably acceptable.

Reply:
It should go from `fromIndex + 1` because the first one is already in the result, but now that I'm reading it, that won't check if it hasn't been used in the phrase before, will adjust...